### DISCUSSION TITLE
How to remove Snippet Section indent?

### DISCUSSION THREAD

**Discussion by maciejwalkowiak** [facelessuser/pymdown-extensions#1913] [Q&A] [answered]
I have a following piece of code in file `DynamoDbEnhancedClientTests.java`:

```java
public class DynamoDbEnhancedClientTests {

    @Test
    void createsClient() {
        // --8<-- [start:func]
        DynamoDbClient ddb = DynamoDbClient.builder()
                .region(Region.US_EAST_1)
                .build();
        // --8<-- [end:func]
    }
}
```

Now when I import `func` snippet:

```java
--8<-- "DynamoDbEnhancedClientTests.java:func"
```

It gets rendered with exactly the same indentation as in the source file:

![image](https://user-images.githubusercontent.com/1357927/213559035-bce974a3-14a1-4fee-b5ae-d96fed445d88.png)

Is there an option to reduce the indentation (by the length of the indent in the first line) in each line so that the code is rendered like this:

![image](https://user-images.githubusercontent.com/1357927/213559589-ef76e743-375f-4942-9609-230cbe1d7de2.png)

Any help appreciated!

**Comment by facelessuser** [comment, marked as answer]
There is not currently a way to do so. Snippets is a fairly simple approach to injecting content from one file into another. Constant requests to make it a little more advanced have led us to sections 🙂. Sections are a fairly new feature. We could potentially consider adding some sort of dedent functionality, but at this time, I'm not quite sure what the syntax would even look like to make that happen...under the hood, it would be fairly easy to do so.

Now, snippets can be complex and even have nested snippets. If I were to add any kind of dedent functionality, I don't imagine I want to get too complicated and try to consider nested snippet context when dedenting, but if we are talking about a simple trigger to make a snippet get dedented, then I may be open to it.

**Reply by maciejwalkowiak** [reply]
Thank you for quick reply. I am new to Python Markdown world so I won't try to suggest how/if this should be done. Currently I solved it with an extra Javascript which is fine for my use case. For the reference:

```javascript
const nodes = document.querySelectorAll("code")

nodes.forEach(node => {
    // skip nodes without multiple lines
    if (node.children.length > 1) {
        // [0] contains a line number
        // [1] contains first line of code
        let firstCodeLine = node.children[1];
        const hasPad = firstCodeLine.innerHTML.trim().length === 0
        // remove pad only from ones that have empty space before the code start
        if (hasPad) {
            const padLength = firstCodeLine.innerHTML.length;
            const paddingNodes = node.querySelectorAll("a + span.w")
            // remove pad from each line
            paddingNodes.forEach(pn => {
                pn.innerHTML = pn.innerHTML.substring(padLength);
            })
        }
    }
})
```

Then I include it in `mkdocs.yml`:

```yml
extra_javascript:
  - javascripts/extra.js
```

**Comment by chehsunliu** [comment]
I need this feature too. I just migrated to Material for MkDocs from [Asciidoctor](https://asciidoctor.org/) and [docsify](https://docsify.js.org/#/). They both support the dedent functionality. For now, I have to do some workaround.

**Reply by facelessuser** [reply]
There is no current planned path to resolve this at this time. Feel free to create a feature request. We will at least consider it as a possibility for the future.
